obelisk/cpp/obelisk_cpp/include/obelisk_controller.h

```diff
@@ -0,0 +1,101 @@
+#pragma once
+
+#include "obelisk_node.h"
+
+namespace obelisk {
+
+    template <typename ControlMessageT, typename EstimatorMessageT> class ObeliskController : public ObeliskNode {
+      public:
+        explicit ObeliskController(const std::string& name, const std::string& ctrl_key = "publisher_ctrl",
+                                   const std::string& est_key   = "subscriber_est",
+                                   const std::string& timer_key = "timer_ctrl")
+            : ObeliskNode(name), ctrl_key_(ctrl_key), est_key_(est_key), timer_key_(timer_key) {
+
+            // Register all components
+            this->RegisterTimer("timer_ctrl_setting", timer_key_, std::bind(&ObeliskController::ComputeControl, this));
+            this->RegisterPublisher<ControlMessageT>("pub_ctrl_setting", ctrl_key_);
+            this->RegisterSubscription<EstimatorMessageT>(
+                "sub_est_setting", est_key_, std::bind(&ObeliskController::UpdateXHat, this, std::placeholders::_1));
+        }
+
+        /**
+         * @brief configures the node.
```

Comment:
open another issue for us to clean the docstrings on both branches. we should sync the docstrings up, capitalize the first letter, use correct punctuation, etc.